sample-clients/csharp/client/ScoringClient.cs

```diff
@@ -47,20 +48,46 @@ public ScoringClient(string host, int port, bool useSsl = false, string authKey
 
         public async Task<T> ScoreAsync<T>(IScoringRequest request) where T : class
         {
-            var result = await _client.PredictAsync(request.MakePredictRequest());
-            return result.Outputs["output_alias"].Convert<T>();
+            var predictRequest = request.MakePredictRequest();
+
+            return await RetryAsync(async () => {
+                var result = await _client.PredictAsync(predictRequest);
+                return result.Outputs["output_alias"].Convert<T>();
+            });
         }
 
-        public float[] Score(IScoringRequest request)
+        private static async Task<T> RetryAsync<T>(
+            Func<Task<T>> operation, int retryCount = 20,
+            int initialDelayInMs = 500, int maxDelayInMs = 10000)
         {
-            return Score<float[]>(request);
+            var delay = initialDelayInMs;
+
+            while (true)
+            {
+                try
+                {
+                    return await operation();
+                }
+                catch (RpcException rpcException)
+                {
+                    if (!IsTransient(rpcException) || --retryCount <= 0)
+                    {
+                        throw;
+                    }
+
+                    await Task.Delay(Math.Min(delay, maxDelayInMs));
+                    delay *= 2;
```

Comment:
I think this is cleaner:
await Task.Delay(delay);
delay = Math.Min(2*delay, maxDelayInMs);

because there is no chance of overflow